extensions/sendbird/v1/actions/createUser/createUser.ts

```diff
@@ -84,3 +87,19 @@ export const createUser: Action<typeof fields, typeof settings> = {
     }
   },
 }
+
+const parseNickname = (payload: any): string => {
+  const nickname: string = payload.fields.nickname
+  if (!isEmpty(nickname)) return nickname
+
+  const firstName = payload.patient.profile?.first_name
+  const lastName = payload.patient.profile?.last_name
+
+  if (isEmpty(nickname) && isEmpty(firstName) && isEmpty(lastName)) {
+    throw new Error(
+      'Nickname is not specified, and both first name and last name are unknown.'
+    )
+  }
+
+  return `${firstName as string} ${lastName as string}`
```

Comment:
If only firstName is undefined and last name is Samuels, for example, this would return `undefined Samuels`

Correct logic would be
- use nickname if present
- use firstName lastName if both present
- use neither if either firstName or lastName is not present (i.e. do not set nickname -- let the consumer handle this)

@michal-grzelak the error thrown would cause the activity to fail / be unsuccessful during Orchestration yes? If so, then we do not throw an error for something like this -- it's a convenience that we are implementing, and should fail gracefully

Reply:
@kubiak85 make sure the tests are updated to reflect all the possible scenarios please

Reply:
(and to be clear, you followed the functional requirements written in the ticket description correctly, so this is not an error on your part. the ticket wasn't clear enough and missed describing what to do in some scenarios)

Reply:
@rahulkeerthi Yes it would make the action fail, so I believe the throw should be removed, so that the nickname is assigned in the scenarios you described above and fall back to just `undefined` in every other case.

Reply:
If sending `nickname: undefined` is understood by Sendbird as set value to nothing, then good, otherwise omit nickname field altogether